socorro/external/es/super_search_fields.py

```diff
@@ -176,39 +171,71 @@ def update_field(self, **kwargs):
                 'be created before it can be updated. ' % params['name']
             )
 
-        if 'storage_mapping' in params and old_value['storage_mapping']:
+        # Then, if necessary, verify the new mapping.
+        if (
+            (
+                'storage_mapping' in params
+                and params['storage_mapping'] != old_value['storage_mapping']
+            ) or (
+                'in_database_name' in params
+                and params['in_database_name'] != old_value['in_database_name']
+            )
+        ):
+            # This is a change that will have an impact on the Elasticsearch
+            # mapping, we first need to make sure it doesn't break.
+            new_mapping = self.get_mapping(overwrite_mapping=params)
+
+            # Try the mapping. If there is an error, an exception will be
+            # raised. If an exception is raised, the new mapping will be
+            # rejected.
+            self.test_mapping(new_mapping)
+
+        if (
+            'storage_mapping' in params
+            and params['storage_mapping'] != old_value['storage_mapping']
+        ):
             # The storage mapping is an object, and thus is treated
             # differently than other fields by Elasticsearch. If a user
             # changes the object by removing a field from it, that field won't
             # be removed as part of the update (which performs a merge of all
-            # objects in the back-end). We therefore want to remove that field
-            # before we do the merge, so that it is entirely overwritten.
-            es_connection.update(
-                index=self.config.elasticsearch.elasticsearch_default_index,
-                doc_type='supersearch_fields',
-                body={'script': 'ctx._source.remove("storage_mapping")'},
-                id=params['name'],
+            # objects in the back-end). We therefore want to perform the merge
+            # ourselves, and remove the field from the database before
+            # re-indexing it.
+            new_doc = old_value.copy()
```

Comment:
The only other time `old_value` is used is when you later reference `old_value['storage_mapping']` but surely that's equal to what it was before.